### PR TITLE
Decode avro cloud event

### LINKED ISSUE #2928: Decode cloudevent with avro binary
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
Kserve [decode method](https://github.com/kserve/kserve/blob/f9e9a079712e202e11855419a03b2966320fe575/python/kserve/kserve/protocol/dataplane.py#L237) does not decodes the cloudevent if it contains avro binary data. It directly passed as cloudevent to preprocess method.

**What did you expect to happen:**
Kserve should decode the avro cloudevent and pass only the bytes data to the preprocess.

**What's the InferenceService yaml:**
[To help us debug please run `kubectl get isvc $name -n $namespace -oyaml` and paste the output]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version:
- Knative Version:
- KServe Version: 
- Kubeflow version:
- Cloud Environment:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube/Kind version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2928 

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
Cloud event with avro binary data no longer passed as cloudevent to the preprocess method. Now on only the bytes data is passed.
```

### CODE REVIEW

File: python/kserve/kserve/protocol/dataplane.py [hunk 1]
```diff
@@ -238,18 +238,25 @@ def decode_cloudevent(self, body) -> Tuple[Union[Dict, InferRequest], Dict]:
...
-                decoded_body = orjson.loads(body.data.decode('UTF-8'))
-                attributes = body._get_attributes()
-            except (orjson.JSONDecodeError, UnicodeDecodeError) as e:
-                # If decoding or parsing failed, check if it was supposed to be JSON UTF-8
-                if "content-type" in body._attributes and \
-                        (body._attributes["content-type"] == "application/cloudevents+json" or
-                         body._attributes["content-type"] == "application/json"):
-                    raise InvalidInput(f"Failed to decode or parse binary json cloudevent: {e}")
+            attributes = body._get_attributes()
+            avro_content_types = ("application/avro", "application/cloudevents+avro", "application/x-avro-binary",
```
Comment (yuzisun) [inline python/kserve/kserve/protocol/dataplane.py]: I think the content type is not limited to avro, it is for other binary encoded types as well such as protobuf. see more details on this issue https://github.com/cloudevents/spec/issues/520

File: python/kserve/test/test_server.py [hunk 2]
```diff
@@ -146,13 +145,7 @@ def _parserequest(self, request):
         return record1
 
     def preprocess(self, request, headers: Dict[str, str] = None):
-        if isinstance(request, CloudEvent):
```
Comment (yuzisun) [inline python/kserve/test/test_server.py:149]: I think the cloudevent attributes are passed as headers can we assert that instead ?

Comment (sivanantha321) [reply python/kserve/test/test_server.py:149]: done

**Comment by yuzisun** [comment]
Thanks @sivanantha321 !
/lgtm
/approve

**Pull request merged**